core/trashcan.js

```diff
@@ -470,7 +470,8 @@ Blockly.Trashcan.prototype.onDelete_ = function() {
     if (trashcan.workspace_.options.maxTrashcanContents <= 0) {
       return;
     }
-    if (event.type == Blockly.Events.BLOCK_DELETE) {
+    if (event.type == Blockly.Events.BLOCK_DELETE &&
+        event.oldXml.tagName == 'shadow') {
```

Comment:
This actually blocks everything. You'll want to convert the tag name to lower case and make it != 'shadow'.

Reply:
Ok actually fixed, and retested (with shadow blocks and non-shadow blocks this time).

Making a note to add trashcan unit tests, because this is just unacceptable.